File: Stack/0_U_linkedlist.py
# Why to use linked list in stack
#-> size of stack is not known in advance
class Node:
    def __init__(self,data) -> None:
        self.data = data
        self.next = Node
    
class Stack:
    def __init__(self) -> None:
        self.head = None
    
    def push(self, data):
        if self.head ==None:
            self.head=Node(data)
        else:
            newnode =Node(data)
            newnode.next=self.head
            self.head = newnode
    
    def isEmpty(self):
        if self.head==None:
            return True
        else:
            return False

    

    def display(self):
        itnode=self.head
        if self.isEmpty():
            print("stack is underflow")
        else:
            while itnode !=None:
                print(itnode.data, end=" ")
                itnode=itnode.next

    
if __name__=="__main__":
    mystack= Stack()
    mystack.push(1)
    mystack.push(2)
    mystack.push(3)
    mystack.push(4)

    #display element
    mystack.display()

File: Stack/0_U_linkedlist.cpp
#include <bits/stdc++.h>
using namespace std;

class Node{
    public:
        int data;
        Node* link;

        Node(int n){
            this->data=n;
            this->link=NULL;
        }
};
class Stack{
    Node* top;
    public:
    stack(){
        top=NULL;
    }

    void push(int data){
        Node* temp= new Node(data);

        if(!temp){
            cout<<"stack is overflow"<<endl;
            exit(1);
        }
        temp->data=data;
        temp->link = top;
        top= temp;
    }
    void pop(){
        Node* temp;
        if(isEmpty()){
            cout<<"stack is underflow "<<endl;
            exit(1);
        }
        else{
            temp =top;
            top=temp->link;
            free(temp);
        }
    }

    bool isEmpty(){
        return top==NULL;
    }
    int peek(){
        if( !isEmpty())
            return top->data;
        
        else
            exit(1);
        
    }

    void display(){
        Node* temp;
        if(top==NULL){
            cout<<"stack is underflow"<<endl;
            exit(1);
        }
        else{
            temp =top;
            while (temp!=NULL){
                cout<<temp->data;
                temp=temp->link;
                if(temp!= NULL){
                    cout<<"->";
                }
                cout<<endl;
            }
            
            
        }


    }
};
int main(){
    Stack s;
    s.push(11);
    s.push(22);
    s.push(33);
    s.push(44);
    s.push(55);
    //s.display();
    //cout<<"top of stack"<<endl;
    //cout<< s.peek() <<endl;
    //s.pop();
    s.display();

    return 0;
}